packages/phone-number-privacy/src/config.ts

```diff
@@ -1,4 +1,57 @@
-const functions = require('firebase-functions')
-export const functionConfig = functions.config()
+import * as functions from 'firebase-functions'
+import knex from 'knex'
 
-export const PHONE_NUMBER_PRIVACY_SECRET_KEY = functionConfig.envs.secret_key
+const DEV_MODE = process.env.NODE_ENV !== 'production' || process.env.FUNCTIONS_EMULATOR === 'true'
+
+interface Config {
+  salt: {
+    key: string
+  }
+  db: {
+    user: string
+    password: string
+    database: string
+    host: string
+  }
+}
+
+let config: Config
+
+if (DEV_MODE) {
+  console.debug('Running in dev mode')
+  config = {
+    salt: {
+      key: 'fakeSecretKey',
+    },
+    db: {
+      user: 'postgres',
+      password: 'fakePass',
+      database: 'phoneNumberPrivacy',
+      host: 'fakeHost',
+    },
+  }
+} else {
+  const functionConfig = functions.config()
+  config = {
+    salt: {
+      key: functionConfig.salt_key,
+    },
+    db: {
+      user: functionConfig.db_username,
+      password: functionConfig.db_pass,
+      database: functionConfig.db_name,
+      host: `/cloudsql/${functionConfig.db_host}`,
+    },
+  }
+}
+
+export const connectToDatabase = () => {
+  console.debug('Creating knex instance')
+  return knex({
```

Comment:
What are the types of errors that can be thrown when creating this connection?

Reply:
Not sure. Try not to overthink this code here. Just throwing together the skeleton of this conection